app/Services/PopulateService.php

```diff
@@ -0,0 +1,160 @@
+
+                Building::firstOrCreate([
+                    'name' => $values[0],
+                    'description' => $values[1],
+                    'address' => $values[2],
+                    'type_id' => $typeId,
+                    'city_id' => $cityId,
+                    'main_image_url' => $values[5],
+                    'pending' => false
+                ]);
+            }
+
+            fclose($file);
+        } else {
+            die(config('scraper.command.buildings_filename') . ' file was not found');
+        }
+    }
+    private function populateAmenities(string $path): void
+    {
+        $file = fopen($path . config('scraper.command.amenities_filename'), 'r');
+
+        if ($file) {
+            while (($line = fgets($file)) !== false) {
+                $values = explode(';', $line);
+                $buildingName = $values[0];
+                $amenityName = $values[1];
+
+                $amenity = Amenity::firstOrCreate([
+                    'name' => $amenityName
+                ]);
+                $building = Building::where([
+                    'name' => $buildingName
+                ])->firstOrFail();
+
+                $building->amenities()->firstOrCreate([
+                    'id' => $amenity->id
+                ]);
+            }
+
+            fclose($file);
+        } else {
+            die(config('scraper.command.amenities_filename') . ' file was not found');
+        }
+    }
+
+    /**
+     * @throws ModelNotFoundException
+     */
+    private function populateBuildingsImages(string $path): void
+    {
+        $file = fopen($path . config('scraper.command.images_filename'), 'r');
+
+        if ($file) {
+            while (($line = fgets($file)) !== false) {
+                try {
+                    $values = explode(';', $line);
+                    $hotelName = $values[0];
+                    $imageUrl = $values[1];
+
+                    $image = BuildingImage::firstOrNew([
+                        'url' => $imageUrl
+                    ]);
+                    $building = Building::where([
+                        'name' => $hotelName
+                    ])->firstOrFail();
+
+                    $building->images()->save($image);
+                } catch (Exception) {
+                    echo 'Could not find the building record associated with image' . "\n";
+                }
+            }
+            fclose($file);
+        } else {
+            die(config('scraper.command.images_filename') . ' file was not found');
+        }
+    }
+
+    private function getCountry(string $countryName): Country
+    {
+        return Country::firstOrCreate([
+            'name' => $countryName
+        ]);
+    }
+
+    private function getCityId(string $cityName, Country $country): int
+    {
+        $city = $country->cities()->firstOrCreate([
+            'name' => $cityName
+        ]);
+
+        return $city->id;
```

Comment:
I don't know if we want to keep this like that, but this makes IDEs tell you that the `id` property doesn't exist. It is because doing `->cities()` returns a relation, but there is no way for IDE to know the details of this relation.

One solution to this would be adding `/** @var City $city */` before the `$city` variable, but I don't know whether we want to have it like that...